main.py

```diff
@@ -558,13 +562,28 @@ def main(cl_arguments):
     if args.do_full_eval:
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
+        diagnostic_tasks = []
+        target_tasks_remove = []
+        for index, task in enumerate(target_tasks):
+            if task.name in task_modules.ALL_DIAGNOSTICS:
+                diagnostic_tasks.append(task)
+                target_tasks_remove.append(index)
+        # To avoid evaluating twice on a diagnostic task.
+        for index in sorted(target_tasks_remove, reverse=True):
+            del target_tasks[index]
+        if len(diagnostic_tasks) > 0:
+            for task in diagnostic_tasks:
+                ckpt_path = get_best_checkpoint_path(args, "eval", task.name)
+                assert ckpt_path is not None
+                load_model_state(model, ckpt_path, args.cuda, skip_task_models=[], strict=strict)
+                evaluate_and_write(args, model, [task], splits_to_write)
         if args.do_target_task_training or (args.load_model and not args.do_pretrain):
             # If we either do target task training, or if we only evaluate
             # without pretraining or target task training
             # then we evaluate on the target tasks.
             for task in target_tasks:
                 # Find the task-specific best checkpoint to evaluate on.
-                ckpt_path = get_best_checkpoint_path(args, "eval", task.name)
+                ckpt_path = get_best_checkpoint_path(args, "target_train", task.name)
```

Comment:
I don't understand why we use 'eval' checkpoints only if there are diagnostic tasks...

Reply:
:+1:, what's the difference between `eval` and `target_train` ?

Reply:
Actually, this should still be eval not target_train - was doing a slightly different way of testing on diagnostics before the current way. I'll revert the change. 
The difference between eval and target_task is where the function looks for checkpoints.